apps/prairielearn/src/lib/preact-html.ts

```diff
@@ -0,0 +1,12 @@
+import { render } from 'preact-render-to-string/jsx';
+
+import { type HtmlSafeString, unsafeHtml } from '@prairielearn/html';
+import type { VNode } from '@prairielearn/preact-cjs';
+
+// This function must live outside of `preact.tsx` so that it can be imported
+// on the client. `preact.tsx` imports `@prairielearn/compiled-assets`, which
+// cannot be bundled for the browser.
```

Comment:
I forgot why this even needed to be imported on the client in the first place, so I went and checked: this is because I had rewritten some components (e.g. `Scorebar`) into Preact (`ScorebarPreact`) and then implemented the HTML version as `renderHtml(<Scorebar ... />)`.

This may not be the best solution; perhaps we want separate, parallel implementations of components like this? I don't have a strong sense of what the best option is.

Reply:
Done in [269a3c1](https://github.com/PrairieLearn/PrairieLearn/pull/12158/commits/269a3c18d2984f45a285b9092ce7927276537fab).